pysignalscope/scope.py

```diff
@@ -853,7 +921,801 @@ def plot_channels(*channel: List['Scope'], timebase: str = 's', figure_size: Opt
+                # Correct the value
+                HandleScope.shiftstep_x = HandleScope.max_shiftstep_x
+                HandleScope.shift_text_box.set_val(str(HandleScope.shiftstep_x))
+            elif HandleScope.shiftstep_x < HandleScope.min_shiftstep_x:
+                # Shift step x too small
+                logging.info(f"{class_modulename} :Shift step x {HandleScope.shiftstep_x} too small and set to {HandleScope.min_shiftstep_x}")
+                # Correct the value
+                HandleScope.shiftstep_x = HandleScope.min_shiftstep_x
+                HandleScope.shift_text_box.set_val(str(HandleScope.shiftstep_x))
+
+        # Log flow control
+        logging.debug(f"{class_modulename}: Shift step x = {HandleScope.shiftstep_x},Shift step y = {HandleScope.shiftstep_y}")
+
+        # Reset the values
+        HandleScope.last_val = 0
+        HandleScope.shift_slider.set_val(0)
+        # Update the plot
+        HandleScope.shiftfig.canvas.draw_idle()
+
+        # Log flow control
+        logging.debug(f"{class_modulename} :Text is overtaken {HandleScope.shift_dir} (0=x, 1=y).")
+
+    # Callback-Funktion for shift the channel by movement of the slider
+    @staticmethod
+    def shiftchannel(val: float):
+        """
+        Callback-Funktion for shift the channel by movement of the slider.
+
+        This function is private (only for internal usage)
+        Called by mathplotlib-Event and assigned to the slider
+
+        :param val: Current slider position
+        :type val: float
+        """
+        # Check shift direction
+        if HandleScope.shift_dir == 1:
+            delta = val-HandleScope.last_val
+            dshift = (delta*HandleScope.shiftstep_y)
+            HandleScope.shiftlist[HandleScope.chn_index][HandleScope.shift_dir] = \
+                (HandleScope.shiftlist[HandleScope.chn_index][HandleScope.shift_dir]+dshift)
+            # Shift values in y-direction
+            new_y = [value + dshift for value in HandleScope.channelplotlist[HandleScope.chn_index].get_ydata()]
+            # Update dataset
+            HandleScope.channelplotlist[HandleScope.chn_index].set_ydata(new_y)
+        else:
+            delta = val-HandleScope.last_val
+            dshift = (delta*HandleScope.shiftstep_x)
+            shift = HandleScope.shiftlist[HandleScope.chn_index][HandleScope.shift_dir]+dshift
+            # Shift in x-direction
+            new_x = [value + dshift for value in HandleScope.channelplotlist[HandleScope.chn_index].get_xdata()]
+            # Update dataset
+            HandleScope.channelplotlist[HandleScope.chn_index].set_xdata(new_x)
+
+            # Overtake the shift
+            HandleScope.shiftlist[HandleScope.chn_index][HandleScope.shift_dir] = shift
+            # Update the plot
+            HandleScope.shiftfig.canvas.draw_idle()
+
+        # Store current slider value
+        HandleScope.last_val = val
+
+    @staticmethod
+    def reset_slider(event: any):
```

Comment:
the reset button resets only the slider position, but not the already shifted plot.
From the users view, the user would assume pressing `reset` would lead to the initial state (before sliding anything).

Reply:
Indeed, this could be the case. I renamed it to 'Re-calibrate Slider'